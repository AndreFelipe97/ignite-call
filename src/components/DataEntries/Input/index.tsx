import { TextInput } from "@ignite-ui/react";
import { FieldValues, UseFormRegister } from "react-hook-form";

interface InputProps {
  size?: "sm" | "md";
  prefix?: string;
  placeholder?: string;
  name: string;
  register: any;
}

export function Input({
  size = "sm",
  prefix = "",
  placeholder = "",
  name,
  register,
}: InputProps) {
  return (
    <TextInput
      size={size}
      prefix={prefix}
      placeholder={placeholder}
      {...(register && register(name))}
    />
  );
}
