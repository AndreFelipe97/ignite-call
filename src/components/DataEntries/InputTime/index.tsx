import { TextInput } from "@ignite-ui/react";

interface InputProps {
  name: string;
  isDisabled: boolean;
  register: any;
}

export function InputTime({ name, isDisabled, register }: InputProps) {
  return (
    <TextInput
      size="sm"
      type="time"
      step={60}
      disabled={isDisabled}
      data-testid={name}
      {...(register && register(name))}
    />
  );
}
