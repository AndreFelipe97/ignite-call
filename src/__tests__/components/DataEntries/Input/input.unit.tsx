import { render, screen, fireEvent } from "@testing-library/react";
import { Input } from "@/components/DataEntries/Input";

describe("<Input />", () => {
  const register = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render base input", () => {
    render(<Input name="test-input" register={register} />);

    const input = screen.getByRole("textbox");

    expect(input).toBeInTheDocument();
    expect(register).toBeCalledTimes(1);
    expect(input).toMatchSnapshot();
  });

  it("should render correctly with params", () => {
    render(
      <Input
        name="test-input"
        register={register}
        size="md"
        prefix="test-input"
        placeholder="test-input"
      />
    );

    const input = screen.getByPlaceholderText("test-input");

    expect(input).toBeInTheDocument();
    expect(register).toBeCalledTimes(1);
    expect(input).toMatchSnapshot();
  });
});
