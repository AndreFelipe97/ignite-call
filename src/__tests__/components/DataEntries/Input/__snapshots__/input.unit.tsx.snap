// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render base input 1`] = `
<input
  class="c-grKefI"
  placeholder=""
/>
`;

exports[`<Input /> should render correctly with params 1`] = `
<input
  class="c-grKefI"
  placeholder="test-input"
/>
`;
